Receiver/Receiver.cpp

```diff
@@ -0,0 +1,281 @@
+#include "receiver.h"
+#include <map>
+#include<sstream>
+#include<fstream>
+
+
+std::vector<int> DaysInTheMonth = {31,28,31,30,31,30,31,31,30,31,30,31};
+
+Data::Data():date(01),month(01),year(2000),hour(00),minute(00) {}
+
+Data::Data(int dt, int mth, int yr , int hr, int min): date(dt), month(mth), year(yr), hour(hr), minute(min){}
+
+int Data::getHour()
+{
+	return this->hour;
+}
+
+int Data::getDate()
+{
+	return this->date;
+}
+
+/*int Data::getMonth()
+{
+	return this->month;
+}
+int Data::getYear()
+{
+	return this->year;
+}
+int Data::getMinute()
+{
+	return this->minute;
+}
+
+int Receiver::getDays()
+{
+	return this->days;
+}
+int Receiver::getHours()
+{
+	return this->hours;
+}
+*/
+
+Receiver::Receiver():hours(24),days(31) 
+{
+	for (int i = 0; i < hours+1; i++)
+	{
+
+		vector<int> temp_hours(days, 0);
+		this->visitCount.push_back(temp_hours);
+	}
+}
+
+Receiver::Receiver(int hour,int noofDays):hours(hour),days(noofDays)
+{
+	for (int i = 0; i < hours+1; i++)
+	{
+		vector<int> temp_hours(days,0);
+		this->visitCount.push_back(temp_hours);
+	}
+}
+
+
+int checkMonth(int month)
+{
+	if (month < 1 || month>12)
+	{
+		return 0;
+	}
+	/*if (leapYear(BufferFootfall[0].year))
+	{
+		return DaysInTheMonthForLeapYear[month];
+	}*/
+	return DaysInTheMonth[month - 1];
```

Comment:
before accessing an index of array add a `defensive check` to avoid array-overbound issues.